build/Common.props

```diff
@@ -43,7 +43,7 @@
     <StackExchangeRedisPkgVer>[2.1.58,3.0)</StackExchangeRedisPkgVer>
     <StyleCopAnalyzersPkgVer>[1.2.0-beta.354,2.0)</StyleCopAnalyzersPkgVer>
     <SystemCollectionsImmutablePkgVer>1.4.0</SystemCollectionsImmutablePkgVer>
-    <SystemDiagnosticSourcePkgVer>6.0.0</SystemDiagnosticSourcePkgVer>
+    <SystemDiagnosticSourcePkgVer>[6.0.0,8.0)</SystemDiagnosticSourcePkgVer>
```

Comment:
Oh and I guess my prior comment https://github.com/open-telemetry/opentelemetry-dotnet/pull/3605#discussion_r956503471 assumed a world where we do not introduce this upper bound. I guess I'm still open to this option instead, but it is the more restrictive option. Maybe that's good?

Reply:
I think probing the underlying library before emitting IL, and if the probing failed, do the granular/ultimate fallback is something we should do. And once it's done, we won't have the upper bound limit for this particular case, but upper bound is still useful for other cases that we haven't seen yet. I think it makes sense for .NET 8 users to upgrade and avoid 1.3.x version of OpenTelemetry .NET components.

Reply:
Making my point explicit, I don't have strong opinion here, either way is fine with me 👍

Reply:
My thoughts are... we have already removed all of this stuff from main so let's not throw good money after bad 🤣 Meaning, I don't see a lot of value in future proofing something that doesn't have a future. Let's cap 1.3.x line <= 7 which we know works and then if users want 8+ they can bump to 1.4.x+?

Reply:
Yup, agreed. Forcing function to move users to 1.4.x+ makes sense.